docs/architecture/application/graphql.md

```diff
@@ -37,3 +38,204 @@ app = OrchestratorCore(base_settings=AppSettings())
 # register SUBSCRIPTION_MODEL_REGISTRY
 app.register_graphql(query=NewQuery)
 ```
+
+## Domain Models Auto Registration for GraphQL
+
+When using the register_graphql function, all products in the SUBSCRIPTION_MODEL_REGISTRY will be automatically converted into GraphQL types.
+The registration process iterates through the list, starting from the deepest product block and working its way back up to the product level.
+
+However, there is a potential issue when dealing with a ProductBlock that references itself, as it can lead to an infinite loop.
+To handle this situation, you must manually create the GraphQL type for that ProductBlock and add it to the GRAPHQL_MODELS list.
+
+Here's an example of how to do it:
+
+```python
+import strawberry
```

Comment:
Maybe add a comment in this codeblock 

```
# product_block_file.py
```

So that it's clear that the strawberry.lazy call later on references to itself?